### PR TITLE
Cleanup SerdeResolverUtils

### LINKED ISSUE #2370: SerdeResolver cleanup
This is a follow on ticket to make some minor improvements around the Serdes resolver that were highlighted during the initial PR #2367

- [x] `SerdeResolverUtils`
    - [x] invert the if (short circuit) 
    - [x] slightly change how the "is default serdes" is implemented
    - [x] take [this comment](https://github.com/spring-cloud/spring-cloud-stream/pull/2367#discussion_r862886550) into consideration as well
    - [x] Consider using [ObjectProvider](https://github.com/spring-cloud/spring-cloud-stream/pull/2367#discussion_r862892496) to lookup serdes

- [x] `SerdeResolverUtilsTest`
    - [x] add a more complicated Serde type that has 2 level generics
    - [x] add test for edge cases

- [x] ~Reduce duplication in [SerdesProvidedAsBeansTests](
https://github.com/spring-cloud/spring-cloud-stream/pull/2367#discussion_r862580900)~ (chose not to do this as its only repeated 2x. If it were 3x I would factor out the commonality)

### PR DESCRIPTION
**Pull request by onobc**
* Refactor SerdeResolverUtils logic
* Add more tests in SerdeResolverUtilsTests
* Fixed some random test warnings

Fixes #2370

### CODE REVIEW

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java [hunk 1]
```diff
@@ -207,7 +207,7 @@ public Serde<?> getOutboundValueSerde(ProducerProperties producerProperties,
 	 * @return {@link Serde} for the state store key.
 	 */
 	public Serde<?> getStateStoreKeySerde(String keySerdeString) {
-		return getKeySerde(keySerdeString, (Map<String, ?>) null);
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java:210]: **[UNRELATED-WARNING-FIX]**

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java [hunk 2]
```diff
@@ -217,7 +217,7 @@ public Serde<?> getStateStoreKeySerde(String keySerdeString) {
 	 */
 	public Serde<?> getStateStoreValueSerde(String valueSerdeString) {
 		try {
-			return getValueSerde(valueSerdeString, (Map<String, ?>) null);
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java:220]: **[UNRELATED-WARNING-FIX]**

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java [hunk 3]
```diff
@@ -242,20 +242,18 @@ private Serde<?> getKeySerde(String keySerdeString, Map<String, ?> extendedConfi
 	}
 
 	private Serde<?> getKeySerde(String keySerdeString, ResolvableType resolvableType, Map<String, ?> extendedConfiguration) {
-		Serde<?> keySerde = null;
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java:245]: **[UNRELATED-WARNING-FIX]**

Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KeyValueSerdeResolver.java:245]: **[UNRELATED-WARNING-FIX]**

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 4]
```diff
@@ -49,118 +48,142 @@ abstract class SerdeResolverUtils {
...
 		}
 		List<SerdeWithSpecificityScore> matchingSerdes = new ArrayList<>();
-
-		context.getBeansOfType(Serde.class).forEach((beanName, serdeBean) -> {
-			final Class<?> beanConfigClass = ClassUtils.resolveClassName(((AnnotatedBeanDefinition)
-					context.getBeanFactory().getBeanDefinition(beanName))
-					.getMetadata().getClassName(),
-				ClassUtils.getDefaultClassLoader());
+		ResolvableType serdeType = ResolvableType.forClassWithGenerics(Serde.class, targetType);
+		String[] serdeBeanNames = BeanFactoryUtils.beanNamesForTypeIncludingAncestors(context.getBeanFactory(),
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:173]: @artembilan I was not able to use `context.getBeanProvider` as it did not handle sorting (it does respect `@Order` but that will not work here as the order is dependent on the target type, not the bean definition) - it otherwise returns them in order they are defined in the config class. However, I was able to use similar `BeanFactoryUtils.beanNamesForTypeIncludingAncestors` to get the proper bean names and then grab the bean and bean definition so I can gets its actual type and sort by it. 

This let me simplify things 2 ways:
1. no reflection on the config class to get the return type of the bean method (I use the bean definition for this now)
2. no need to be the final arbiter w/ manual call of `serdeBeanGeneric.isAssignableFrom` (the `beanNamesForTypeIncludingAncestors` handles all of that internally now).

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java [hunk 5]
```diff
@@ -17,76 +17,355 @@
...
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.mock;
 
 /**
  * Unit tests for {@link SerdeResolverUtils}.
  *
  * @author Chris Bono
  */
+@SuppressWarnings({ "rawtypes", "NewClassNamingConvention", "unchecked" })
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java:50]: These are expected as we do all sort of different types of generic combos in this test.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java [hunk 6]
```diff
@@ -17,76 +17,355 @@
...
 class SerdeResolverUtilsTests {
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java:51]: I ❤️ using nested in cases like this as it helps organize the tests logically. Take a look at the output in IDEA when I run this test suite:
Screen Shot 2022-05-11 at 5 05 18 PM

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 7]
```diff
@@ -49,118 +48,142 @@ abstract class SerdeResolverUtils {
...
-						matchingSerdes.add(new SerdeWithSpecificityScore(calculateScore(targetType, serdeBeanGeneric), serdeBean));
-					}
-				});
-			}
-			catch (Exception e) {
-				if (LOG.isTraceEnabled()) {
-					LOG.trace("Failed to introspect Serde bean method '" + serdeBean + "'", e);
+				BeanDefinition beanDefinition = context.getBeanFactory().getMergedBeanDefinition(beanName);
+				ResolvableType serdeBeanGeneric = beanDefinition.getResolvableType().getGeneric(0);
+				Serde<?> serdeBean = context.getBean(beanName, Serde.class);
```
[the remark this answers is not in the document]
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:179]: Yes, this thought crossed my mind as well as sounded something like this....

"We are not even using this ordered list of matching beans. Now we have this complicated set of tests to look at / maintain. We should only add things when we actually need them".   

1. However, we do have to go through them all in order to find the "closest/best match".
2. I wanted to test the sort/ordering from above so I ended up w/ the test permutations.

Since at this point the code was already there I chose to leave it in rather than pare it back. To limit any public use of the N-based API we could instead create a ` public Serde<?> findClosestMatchingSerde` API which delegates to the existing one and move the existing one to package-protected. 

I think this would be a good thing. Wdyt?

[the remark this answers is not in the document]
Comment (artembilan) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:179]: Not sure how the suggested refactoring is relevant to my concern about beans loading overhead.

According the rest of your logic, you do propagate `Serde` beans in the `SerdeWithSpecificityScore`, but you do not use them in the score calculation and then you don't use the in compare function.
Not sure how dropping bean reference at this point in favor if just its name may break all your tests...

**Review by artembilan — CHANGES_REQUESTED** [review]

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:163]: Ahh ok. I misunderstood the concern.

Yes, since we do not actually use the bean for anything other than returning to the caller, we could instead just return the beanName(s) and let the caller load the bean(s), OR if we pared the API down to return only a single "top match" we could just load that single serde bean.

One thing to remember is that this operation is during the initial configuration so not loading the handful of serde beans probably configured will not be a big performance gain.

> Not sure how dropping bean reference at this point in favor if just its name may break all your tests..

Let's not worry about that. I will update the tests based on the direction we choose. (we tell them what to do ;P )

**Review by sobychacko — APPROVED** [review]

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (artembilan) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:163]: Well, if it's used only on configuration phase, then OK: let's your logic a is.
Plus there is no too much to worry about since the API in the end is not `public`.

**Comment by sobychacko** [comment]
@onobc The PR looks good. I am ok with merging it. We can make any further changes through subsequent changes if need be.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 8]
```diff
@@ -49,118 +48,126 @@ abstract class SerdeResolverUtils {
...
+		}
+		Class<?> serdeClass = serde.getClass();
+		return Serdes.Integer().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Long().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Short().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Double().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Float().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.ByteArray().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.String().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.UUID().getClass().isAssignableFrom(serdeClass);
```
Comment (artembilan) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: This will fall under another SonarCube rule - long boolean expression, but I don't find it very reasonable 😄 

On the other thought I wonder why plain `serde instance Serdes.Integer()` doesn't work instead? Or even `equals()`...
Just because those `Serdes` classes are all `final`, therefore cannot be extension or so:
```
static public final class IntegerSerde extends WrapperSerde<Integer> {
```

**Review by artembilan — CHANGES_REQUESTED** [review]

**Comment by onobc** [comment]
> @onobc The PR looks good. I am ok with merging it. We can make any further changes through subsequent changes if need be.

Sounds good @sobychacko . The changes were straight forward to update though. Let me push the commit and we can decide which API to keep.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 8 above)
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: Yeh, I had the same thoughts about isAssignable here. 

Either way though, I do prefer the concise "long boolean expression" way of doing this as it keeps it easier to read.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 9]
```diff
@@ -49,123 +48,130 @@ abstract class SerdeResolverUtils {
...
+		if (String.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.String();
 		}
-		else {
-			// If the type is Object, then skip assigning the JsonSerde and let the fallback mechanism takes precedence.
-			if (!genericRawClazz.isAssignableFrom((Object.class))) {
-				serde = new JsonSerde(genericRawClazz);
-			}
+		if (UUID.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.UUID();
```
Comment (artembilan) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:123]: How about `Serdes.serdeFrom(Class<T> type)` instead of all of these ifs ?

Then you catch their `IllegalArgumentException` and return `null` (or whatever logic you'd like to do)

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 10]
```diff
@@ -49,123 +48,130 @@ abstract class SerdeResolverUtils {
...
+		}
+		Class<?> serdeClass = serde.getClass();
+		return Serdes.Integer().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Long().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Short().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Double().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Float().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.ByteArray().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.String().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.UUID().getClass().isAssignableFrom(serdeClass); //NOSONAR
```
Comment (artembilan) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: I think this `//NOSONAR` has to be on a first line of the "failed" code block:
https://github.com/spring-projects/spring-integration/blob/main/spring-integration-http/src/main/java/org/springframework/integration/http/multipart/UploadedMultipartFile.java#L67 - direct array argument storage.
thttps://github.com/spring-projects/spring-integration/blob/main/spring-integration-http/src/main/java/org/springframework/integration/http/multipart/UploadedMultipartFile.java#L67 - too long method.

**Review by artembilan — CHANGES_REQUESTED** [review]

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 10 above)
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: Got it - I thought it was at the end of the statement. I will adjust and then if ti fails during Jenkins I will adjust that.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 9 above)
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:123]: Yeh I was originally looking in that class to see if there was some built-in methods that would work for us. They do handle one more type than we do IIRC but I will take a scan at using it.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 10 above)
Comment (artembilan) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: Yeah... I'm not sure there is SonarCube involved in Spring Cloud Stream CI plan.
@sobychacko ?

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 9 above)
Comment (artembilan) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:123]: Sure! Let it be then as it is! Probably not related to this PR anyway.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 10 above)
Comment (artembilan) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: So, @onobc , are you going to move that `//NOSONAR`?
Or let it just live as is?

Comment (sobychacko) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: Yes, @artembilan, there is one on our Jenkins CI plans.

Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:140]: Moving, give me a second. Plus, now I am tweaking around w/ those methods. I will commit shortly.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java [hunk 11]
```diff
@@ -16,77 +16,349 @@
...
+			@MethodSource("kafkaStreamsBuiltInTypes")
+			void returnsStandardSerdeForKafkaStreamsBuiltInType(Class<?> builtInType, Serde<?> expectedBuiltInSerde) {
+				contextRunner.run((context) ->
+					assertThat(SerdeResolverUtils.resolveForType(context, ResolvableType.forClass(builtInType), fallback))
+						.isInstanceOf(expectedBuiltInSerde.getClass()));
+			}
+
+			static Stream<Arguments> kafkaStreamsBuiltInTypes() {
+				return Stream.of(
+					arguments(String.class, Serdes.String()),
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java:95]: I did re-order these and make them match the new standard default classes in SRU

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 12]
```diff
@@ -48,124 +47,120 @@ abstract class SerdeResolverUtils {
 
 	private static final Log LOG = LogFactory.getLog(SerdeResolverUtils.class);
 
+	/** Classnames of the standard built-in Serdes supported in {@link Serdes#serdeFrom(Class)} */
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:50]: I went ahead and squashed/rebased . The changes since last review round are here and the 2 related standard default methods.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 13]
```diff
@@ -48,124 +47,120 @@ abstract class SerdeResolverUtils {
...
-		}
-		else if (Short.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Short();
-		}
-		else if (Double.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Double();
-		}
-		else if (Float.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Float();
+		List<String> matchingSerdes = SerdeResolverUtils.beanNamesForMatchingSerdes(context, targetType);
```
Comment (artembilan) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:87]: Nit: don't need to point to itself if you call your own methods.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java [hunk 14]
```diff
@@ -48,124 +47,120 @@ abstract class SerdeResolverUtils {
...
+		Serdes.String().getClass().getName(),
+		Serdes.Short().getClass().getName(),
+		Serdes.Integer().getClass().getName(),
+		Serdes.Long().getClass().getName(),
+		Serdes.Float().getClass().getName(),
+		Serdes.Double().getClass().getName(),
+		Serdes.ByteArray().getClass().getName(),
+		Serdes.ByteBuffer().getClass().getName(),
+		Serdes.Bytes().getClass().getName(),
+		Serdes.UUID().getClass().getName());
```
Comment (artembilan) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:61]: Why names? Why just classes not enough?
Looks like you have the target class as well to check its existence in this set.
Otherwise you do extra step to take the name of that class.

**Review by artembilan — CHANGES_REQUESTED** [review]

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 14 above)
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:61]: Yep, it is intentional and the reason is fear of holding on to a class reference vs. a reference to a string. Probably paranoia though.

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java (hunk 13 above)
Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:87]: It is definitely a style thing as some Sonar complain w/o reference to it (even when static). However, I had it w/o before and I have no strong preference. Updating...

Comment (artembilan) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java:87]: That sound suspicious: a couple lines below you use a `getStandardDefaultSerde()` without `SerdeResolverUtils.` prefix...

**Review by artembilan — APPROVED** [review]

**Comment by onobc** [comment]
Screen Shot 2022-05-13 at 10 52 00 AM

Finally! Haha just kidding. Great review - thanks for all the feedback @artembilan

**Comment by sobychacko** [comment]
@onobc checkstyle failed on me during the local build. Do you mind fixing it?

**Comment by onobc** [comment]
@sobychacko yep, my bad. I forgot to do final local build after last changes. Adjusting now.

**Comment by onobc** [comment]
Resolved via https://github.com/spring-cloud/spring-cloud-stream/commit/cc05160cc05825914a827c5511503770bde24d4d
